q2_composition/_diff_abundance_plots.py

```diff
@@ -138,7 +146,8 @@ def da_barplot(output_dir: str,
                significance_threshold: float = 1.0,
                effect_size_threshold: float = 0.0,
                feature_ids: qiime2.Metadata = None,
-               level_delimiter: str = None):
+               level_delimiter: str = None,
+               label_limit: Optional[int] = None):
```

Comment:
I don't think `Optional` is necessary here. Have you tried the following:

```suggestion
               label_limit: int = None):
```

Providing a default should make it optional (this is what we do for several of the parameters above).